manifests/init.pp

```diff
@@ -20,58 +20,53 @@
 #
 # To ensure that windows_firwall is running:
 #
-#   class { 'windows_firewall':
-#     ensure => 'running',
-#   }
+#   include ::windows_firewall
 #
 class windows_firewall (
-    $ensure = 'running'
+  String $ensure = 'running',
```

Comment:
I don't like this pattern of using Enum for params that are just used by another resource type like this. The reasoning being that the resource type itself already has logic in it to only accept certain strings and will handle things properly. If the resource adds other available strings, it would not work for us. Basically we duplicate effort already done by the type system. If anything, I think we should use `Variant[Boolean, String]`

Reply:
With Variant, it makes all things more complicated, I think. We have Enum of absent/present in exception.pp, that‘s why I thought it makes sense.